File: src/pkg/runtime/race0.c
// Copyright 2011 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

// Stub implementation of the race detector API.
// +build !race

#include "runtime.h"

uintptr
runtime·raceinit(void)
{
	return 0;
}

void
runtime·racefini(void)
{
}


void
runtime·racemapshadow(void *addr, uintptr size)
{
	USED(addr);
	USED(size);
}

void
runtime·racewritepc(void *addr, void *callpc, void *pc)
{
	USED(addr);
	USED(callpc);
	USED(pc);
}

void
runtime·racereadpc(void *addr, void *callpc, void *pc)
{
	USED(addr);
	USED(callpc);
	USED(pc);
}

void
runtime·racewriterangepc(void *addr, uintptr sz, uintptr step, void *callpc, void *pc)
{
	USED(addr);
	USED(sz);
	USED(step);
	USED(callpc);
	USED(pc);
}

void
runtime·racereadrangepc(void *addr, uintptr sz, uintptr step, void *callpc, void *pc)
{
	USED(addr);
	USED(sz);
	USED(step);
	USED(callpc);
	USED(pc);
}

void
runtime·raceacquire(void *addr)
{
	USED(addr);
}

void
runtime·raceacquireg(G *gp, void *addr)
{
	USED(gp);
	USED(addr);
}

void
runtime·racerelease(void *addr)
{
	USED(addr);
}

void
runtime·racereleaseg(G *gp, void *addr)
{
	USED(gp);
	USED(addr);
}

void
runtime·racereleasemerge(void *addr)
{
	USED(addr);
}

void
runtime·racereleasemergeg(G *gp, void *addr)
{
	USED(gp);
	USED(addr);
}

void
runtime·racefingo(void)
{
}

void
runtime·racemalloc(void *p, uintptr sz, void *pc)
{
	USED(p);
	USED(sz);
	USED(pc);
}

void
runtime·racefree(void *p)
{
	USED(p);
}

uintptr
runtime·racegostart(void *pc)
{
	USED(pc);
	return 0;
}

void
runtime·racegoend(void)
{
}

File: src/pkg/runtime/race0.go
// Copyright 2014 The Go Authors.  All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

// +build !race

// Dummy race detection API, used when not built with -race.

package runtime

import (
	"unsafe"
)

const raceenabled = false

// Because raceenabled is false, none of these functions should be called.

func raceReadObjectPC(t *_type, addr unsafe.Pointer, callerpc, pc uintptr)  { gothrow("race") }
func raceWriteObjectPC(t *_type, addr unsafe.Pointer, callerpc, pc uintptr) { gothrow("race") }
func raceinit()                                                             { gothrow("race") }
func racefini()                                                             { gothrow("race") }
func racemapshadow(addr unsafe.Pointer, size uintptr)                       { gothrow("race") }
func racewritepc(addr unsafe.Pointer, callerpc, pc uintptr)                 { gothrow("race") }
func racereadpc(addr unsafe.Pointer, callerpc, pc uintptr)                  { gothrow("race") }
func racereadrangepc(addr unsafe.Pointer, sz, callerpc, pc uintptr)         { gothrow("race") }
func racewriterangepc(addr unsafe.Pointer, sz, callerpc, pc uintptr)        { gothrow("race") }
func raceacquire(addr unsafe.Pointer)                                       { gothrow("race") }
func raceacquireg(gp *g, addr unsafe.Pointer)                               { gothrow("race") }
func racerelease(addr unsafe.Pointer)                                       { gothrow("race") }
func racereleaseg(gp *g, addr unsafe.Pointer)                               { gothrow("race") }
func racereleasemerge(addr unsafe.Pointer)                                  { gothrow("race") }
func racereleasemergeg(gp *g, addr unsafe.Pointer)                          { gothrow("race") }
func racefingo()                                                            { gothrow("race") }
func racemalloc(p unsafe.Pointer, sz uintptr)                               { gothrow("race") }
func racegostart(pc uintptr) uintptr                                        { gothrow("race"); return 0 }
func racegoend()                                                            { gothrow("race") }
